src/editor.c

```diff
@@ -1293,23 +1307,92 @@ editor_get_indent_prefs(GeanyEditor *editor)
 static void on_new_line_added(GeanyEditor *editor)
 {
 	ScintillaObject *sci = editor->sci;
-	gint line = sci_get_current_line(sci);
+	const gint line = sci_get_current_line(sci);
+
+	gboolean doHandleBracket = FALSE;
+	gboolean bracketHandled = FALSE;
+	gint posStart;
+	gint posBracketStart;
 
 	/* simple indentation */
-	if (editor->auto_indent)
-	{
+	if (editor->auto_indent) {
+
+		posStart = sci_get_current_position(sci);
+		posBracketStart = posStart-(sci_get_eol_mode(sci) == SC_EOL_CRLF ? 3 : 2);
+		gchar cPrev = sci_get_char_at( sci, posBracketStart);
+		gchar cNext = sci_get_char_at( sci, posStart);
+
 		insert_indent_after_line(editor, line - 1);
+
+		/* Nice bracket handling */
+		if (cPrev == '{' && cNext  == '}') {
+			posStart = sci_get_current_position(sci);
+			doHandleBracket = TRUE;
+		}
 	}
 
 	if (get_project_pref(auto_continue_multiline))
 	{	/* " * " auto completion in multiline C/C++/D/Java comments */
-		auto_multiline(editor, line);
+		if (auto_multiline(editor, line, FALSE) && doHandleBracket) {
```

Comment:
Why do you change multiline comment behavior?

Reply:
The output of
/** <ENTER>int main() {<ENTER>
with auto-close enabled (<ENTER> == pressing enter-key on keyboard) in an c++ document is

with the change ("|" shows the cursor afterwards:
```
/**
 * int main() {
 *     |
 * }
```

without:
```
/**
 * int main() {
|*
}
```
and in original Geany 1.37 on Windows:
```
/**
 * int main() {
 *|})

```